doc/stdlib/dates.rst

```diff
@@ -19,6 +19,10 @@ Dates and Time Types
 
    ``Period`` types represent discrete, human representations of time.
 
+.. data:: CompoundPeriod
```

Comment:
Unfortunately no, it won't auto-populate. I couldn't find any examples of how to get type docstrings to auto-populate at the time of writing. I'll investigate this more tomorrow but I think another PR will be required to fix this https://github.com/JuliaLang/julia/issues/15623#issuecomment-201332161